_posts/2020-02-15-bastion-floating-on-tide.md

```diff
@@ -0,0 +1,327 @@
+---
+layout: post
+title: "Bastion floating on tide"
+author: Written by Jeremy o0Ignition0o Lempereur
+---
+
+## Introduction
+
+// TODO
+
+## Creating a simple tide project
+
+If you haven’t yet, I highly recommend you to install [cargo-edit](https://github.com/killercup/cargo-edit), a tool that will allow you to add dependencies to a project without the need to edit Cargo.toml (among other really cool features). If you don’t want to install it, you need to skip the `cargo add` statements in the example, and edit Cargo.toml to add the relevant crates as we move along.
+
+```bash
+$ cargo install cargo-edit
+```
+
+Let’s start with a tide example:
+
+```bash
+$ cargo new bastion_with_tide # Create a new project
+$ cd bastion_with_tide # Go to the project folder
+$ cargo add async-std # Add async-std
+$ cargo add tide # Add tide
+```
+
+Now that we created a project and added the required dependencies, let's edit the `src/main.rs` file:
+
+```rust
+fn main() -> Result<(), std::io::Error> {
+    use async_std::task;
+    task::block_on(async {
+        let mut app = tide::new();
+        app.at("/").get(|_| async move { "Hello, world!" });
+        app.listen("127.0.0.1:8080").await?;
+        Ok(())
+    })
+}
+```
+And run the application:
+
+```bash
+cargo run
+[…]
+   Compiling bastion_with_tide v0.1.0 (/Users/ignition/Projects/oss/bastion_with_tide)
+    Finished dev [unoptimized + debuginfo] target(s) in 35.37s
+     Running `target/debug/bastion_with_tide`
+```
+
+Now let’s open a browser and go to [http://127.0.0.1:8080](http://127.0.0.1:8080)
+
+![Tide hello world!](/images/posts/tide-hello-world.png){: .center }
+
+Ok we now have a tide web server up and running, but we’re not doing too much for now. Let’s try to build an application that will do some heavy computing behind the scenes.
+
+### Let's use some CPU cycles!
+
+<s>For the purpose of this example, we will create a fibonacci adder.</s>
+
+Florian skade Gilcher suggested an amazing example, that some of you might have read about already.
+For the purpose of this example, we will create an amazing service for delivering prime numbers on the internet! 
+
+If this exercise doesn't ring a bell, don't worry, I didn't know about it either, and I had an amazing journey discovering Programming Erlang, by [Joe Armstrong](https://en.wikipedia.org/wiki/Joe_Armstrong_(programmer)).
+
+If you were interested in a simple fibonacci service using tide, head over to [the tide examples](https://github.com/http-rs/tide/tree/master/examples), it will probably be available by the time this article is released.
+
+Our prime number service will, given a number of digits, return a prime number with the expected number of digits.
+
+In pseudocode it would look like this:
+
+```
+GET: /prime/digits => return prime_number(digits) 
```

Comment:
maybe `/prime/:digits` I think the `:foo` notation is fairly common for "URL parameters"?